### PR TITLE
Some logical and copy/paste issues found fixed.

### PR DESCRIPTION
**Pull request by Carmot**
Please check some changes because what could be reasonable from a logical point of view breaks the expected functionality desired by the software.
I also saw several issues in 3rd party libraries code but they should be fixed in each respective library repo.

### CODE REVIEW

**Comment by Carmot** [comment]
Travis failed 1 out of 2 due to some linking issues. Please can you check?

**Comment by hugoam** [comment]
The linking issue is random, but I know it doesnt come from your changes so it won't be a problem. (But you can restart the build and it should work). I'll check your fixes when I get home.

File: src/ui/Node.cpp [hunk 1]
```diff
@@ -292,7 +292,7 @@ namespace ui
 		NodeConnection connection = {};
 		
 		CanvasConnect& connect = canvas.m_connect;
-		if(connect.m_origin)
```
[the remark this answers is not in the document]
Comment (Carmot) [reply src/ui/Node.cpp]: In this case connect.m_in and connect.m_out are used without being checked against nullptr.

File: src/ui/Edit/TypeIn.cpp [hunk 2]
```diff
@@ -817,9 +817,8 @@ namespace mud
 			{
 				if(current_line && selection.m_start == selection.m_end)
 				{
-					bool focused = false;
```
[the remark this answers is not in the document]
Comment (Carmot) [reply src/ui/Edit/TypeIn.cpp]: OK. I'll remove this change.

File: src/obj/Proto.cpp [hunk 3]
```diff
@@ -14,6 +14,6 @@ namespace mud
 	{
 		for(Type* type : parts)
 			this->add_part(*type);
-		this->add_part(*this);
```
[the remark this answers is not in the document]
Comment (Carmot) [reply src/obj/Proto.cpp]: It is a suspicious behavior to pass itself as a parameter to its own method.

File: src/geom/Mesh.cpp [hunk 4]
```diff
@@ -104,8 +104,7 @@ namespace mud
 		{
```
[the remark this answers is not in the document]
Comment (Carmot) [reply src/geom/Mesh.cpp]: OK. I'll remove this change.

**Pull request closed without merging**